packages/x-date-pickers/src/internals/hooks/validation/useDateValidation.ts

```diff
@@ -41,42 +18,76 @@ export type DateValidationError =
   | 'maxDate'
   | null;
 
-export const validateDate: Validator<any, DateValidationProps<any, any>> = (
-  utils,
+export const validateDate: Validator<any, DateValidationProps<any, any>> = ({
+  props,
   value,
-  { disablePast, disableFuture, minDate, maxDate, shouldDisableDate },
-): DateValidationError => {
-  const now = utils.date()!;
-  const date = utils.date(value);
+  adapter,
+}): DateValidationError => {
+  const now = adapter.utils.date()!;
+  const date = adapter.utils.date(value);
+
+  const {
+    shouldDisableDate,
+    minDate = adapter.defaultDates.minDate,
+    maxDate = adapter.defaultDates.maxDate,
+    disableFuture,
+    disablePast,
+  } = props;
 
   if (date === null) {
     return null;
   }
 
   switch (true) {
-    case !utils.isValid(value):
+    case !adapter.utils.isValid(value):
       return 'invalidDate';
 
     case Boolean(shouldDisableDate && shouldDisableDate(date)):
       return 'shouldDisableDate';
 
-    case Boolean(disableFuture && utils.isAfterDay(date, now)):
+    case Boolean(disableFuture && adapter.utils.isAfterDay(date, now)):
       return 'disableFuture';
 
-    case Boolean(disablePast && utils.isBeforeDay(date, now)):
+    case Boolean(disablePast && adapter.utils.isBeforeDay(date, now)):
       return 'disablePast';
 
-    case Boolean(minDate && utils.isBeforeDay(date, minDate)):
+    case Boolean(minDate && adapter.utils.isBeforeDay(date, minDate)):
       return 'minDate';
 
-    case Boolean(maxDate && utils.isAfterDay(date, maxDate)):
+    case Boolean(maxDate && adapter.utils.isAfterDay(date, maxDate)):
       return 'maxDate';
 
     default:
       return null;
   }
 };
 
+export const useIsDayDisabled = <TDate>({
```

Comment:
Should we put in this file all the validations (adding month and year validation)

Such that in this folder we test the edge cases, and in components only do one test to check diasble has been applied to the component

Reply:
Yes I think it would be a good idea to move all date-related validations in the `validation` folder